### PR TITLE
Onesignal

### PR DESCRIPTION
**Pull request by chrispypatt**
- [x] Get Onesignal package installed

- [x]  Setup Onesignal notifications

- [x] Add following logic to onesignal

- [ ] Nearby local notifications

### CODE REVIEW

**Comment by chrispypatt** [comment]
Onesignal Issue on iOS 13 on registering user:
` ERROR: Encountered error during push registration with OneSignal: Error Domain=OneSignalError Code=400 "(null)" UserInfo={returned={
errors = (
"["Identifier invalid format."]"
);
success = 0;
}}`

The issue is:
`On iOS 12.2, the following is sent in OSRequestRegisterUser:


`"identifier": "5c868...............40cdb"
`

while on iOS 13 the following is sent:


` "identifier": "{length=32,bytes=0x5c868...............40cdb}"
`

[Temporary Fix](https://stackoverflow.com/a/9084784/811405) implemented with [this]().

File: lib/pages/tab.dart [hunk 1]
```diff
@@ -206,13 +272,84 @@ class _SavourTabPageState extends State<SavourTabPage> with WidgetsBindingObserv
...
+        OneSignal.shared.setEmail(email: user.email);
+      }
+    }else{
+      Provider.of<NotificationSettings>(context).setNotificationsSetting(false);
+      OneSignal.shared.setSubscription(false);
+    }
+  }
+
+  void _notificationHandler(OSNotificationOpenedResult result){
+    if(result.notification.payload.additionalData.isNotEmpty){
```
Comment (branberry) [inline lib/pages/tab.dart:338]: Could you move the if(!mounted) check outside of each step of the if statement, and place it right before if(result.notification.payload.additionalData.containsKey("deal"))?

File: lib/pages/tab.dart [hunk 2]
```diff
@@ -206,13 +272,84 @@ class _SavourTabPageState extends State<SavourTabPage> with WidgetsBindingObserv
...
+          });
+        }else{
+          if(!mounted) return;
+          this.setState(() {
+            var data = result.notification.payload.additionalData;
+            print("Opened notification with additional data: $data");
+          });
+        }
+      }else{
+        if(!mounted) return;
```
Comment (branberry) [inline lib/pages/tab.dart]: can this else be removed? Or are you using this setState to force a refresh?

Comment (chrispypatt) [reply lib/pages/tab.dart]: This should be removed for now. In the future we may handle an empty notification in app.

File: lib/pages/tab.dart (hunk 1 above)
Comment (chrispypatt) [reply lib/pages/tab.dart:338]: Yes, this will be fixed. Thanks!

**Pull request merged**